### DISCUSSION TITLE
Operator manual: Dagger on Gitlab CI self-hosted runners

### DISCUSSION THREAD

**Discussion by shykes** [dagger/dagger#12518] [Ideas]
Provide a definitive guide on how to run Dagger on Gitlab CI self-hosted runners.

**Comment by vikram-dagger** [comment]
@shykes would this be covered by https://github.com/dagger/dagger/pull/4552/files#diff-bda24df101442335ef1f613988436c0c0e3ebc953407907383d7f0bb31345c4fR238? (Appendix B)

**Comment by shykes** [comment]
User feedback [on discord](https://discord.com/channels/707636530424053791/708371226174685314/1072805973170851850):

> In my customers organisation the DinD setup is (almost) forbidden on Gitlab runners so we are looking for ways to run Dagger without that. We are currently rolling out own runners for our team but maintaining them isn's something we are too keen about but we like Dagger that much that we are biting the bullet 🙂 Hopefully soon there are other options.

**Comment by mircubed** [comment]
how is this issue different from #4544 ?

**Comment by shykes** [comment]
> how is this issue different from #4544 ?

They overlap quite a bit on the gitlab part, but #4544 seems to be about one very specific tutorial that happens to use Gitlab, among other tools. This issue focuses on providing operators with a reference documentation for how to install Dagger on Gitlab runners.

**Comment by morlay** [comment]
Btw, for gitlab ci, dind is not a good choice, specially gitlab ci runner on k8s.

Run with remote engine is better.
https://github.com/dagger/dagger/issues/3939#issuecomment-1433990001

**Comment by schlapzz** [comment]
@morlay totally agree with you. Also, almost all the clients I have worked for, dind is not allowed in their setup.

Ive been wondering what would be the best way to set up gitlab runner for dagger. Right now the only clean solution for me is to install a shell runner and Docker on a VM and connect it to GitLab.
However, I would be happy if there is a solution to use Dagger with Kubernetes runners on OpenShift.
